dist/main/index.js

```diff
@@ -1076,29 +1075,27 @@ function setup() {
         }
         catch (error) {
             core.setFailed(`Action failed with error: ${error}`);
-            throw (error);
         }
     });
 }
 function upload() {
     return __awaiter(this, void 0, void 0, function* () {
+        core.startGroup('Cachix: upload');
```

Comment:
Not sure about this: I wouldn't expect Upload to be logged when it's not uploading (also it's called pushing)

Reply:
The issue I was trying to avoid is `core.stopGroup` not being called on exception.

I'm thinking of this as a category, and then the actions inside would vary if there is something to upload or not. I could rename it `push` so it doesn't sound like it's actually pushing?

Reply:
sgtm!

Reply:
Renamed to `Cachix: push`